.github/CODESTYLE.md

```diff
@@ -0,0 +1,305 @@
+* Write code that documents itself. Try to make code as clear as possible (“self-documenting”) to avoid use of comments and redundancy between the two which needs synchronized or is confusingly out of sync. Writing self-documenting code includes using variable names with obvious meaning and documenting ambiguities in variable names when they exist. Minimize commenting “what” (repeating the code with comments) but do it when it is necessary for interpretation.
+* Document why (not what). Algorithmic choices are often the hardest thing to perceive, not function calls on variables.
+* Indent comments to the indentation level of the code which is being commented.
+* Comments shall be written in English with good spelling, punctuation, and grammar.
+* Documentation shall be placed in the code and FORD will be used to generate documentation.
+* TODO: Comments used to remind developers of future or unfinished actions in source code shall begin with “TODO FML” (first middle last initials, as available), and describe the action to be taken. TODO comments shall also list a specific date or event by which the TODO action will be completed.
+
+```fortran
+! TODO JLM : Rename the variable foo to discharge. Target Date: 12/25/20
+
+! TODO JLM : Rename the variable foo to discharge.
+! TODO JLM : Target Date: 12/25/20
+```
+
+* DEPRECATED: Flagging deprecated code sections shall be done in the following way for automated removal at end-of-version code release.
+
+```fortran
+!DEPRECATED >>>
+!code
+!code
+!<<< DEPRECATED
+```
+
+# File header comments
+* Each source file shall contain one file header comment.
+* File header comments shall contain the following:
+  * A concise description of the collective purpose of file contents.
+      * Exception: When a file contains only one class or function definition, this description shall simply state the file is an implementation or definition file for class/function `<name>`. The class or function description will be written into the class or function header comment.
+* Organization name: National Center for Atmospheric Research
+* Current maintainer
+* Author names. Multiple authors should be listed when more than one developer has worked on a source file over time.
+
+```fortran
+! module_super_foo.F
+! Purpose: This module file contains the derived type and
+!          methods for the super_foo class.
+! National Center for Atmospheric Research
+! Responsible: James L McCreight <jamesmcc>ucar<edu>
+! Authors: James McCreight, Logan Karsten, Wei Yu
+```
+
+# Module/Subroutines/Function Usage
+* Always use modules.
+* Intent: all arguments should be given an intent (in,out, inout),
+* Each argument is defined its own line. Document details in- or below- line using FORD (https://github.com/cmacmackin/ford/wiki/Writing-Documentation) conventions.
```

Comment:
Make this into a markdown link with `[text](url)` ?

Reply:
done